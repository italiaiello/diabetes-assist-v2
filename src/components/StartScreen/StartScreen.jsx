import React from 'react'
import Logo from '../../images/DA_logo.svg'

const StartScreen = props => {
    return (
        <article id="startScreen">
            <figure id="startScreenLogo">
                <img src={Logo} alt="Diabetes Assist Logo" />
                <p>DiabetesAssist</p>
            </figure>
            <button className="formButton" onClick={() => props.onRouteChange('signin')}>Get Started</button>
        </article>
    )
}

export default StartScreen