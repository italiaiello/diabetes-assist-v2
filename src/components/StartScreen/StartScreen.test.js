import { shallow } from 'enzyme';
import React  from 'react';
import StartScreen from './StartScreen';

let route = ""

const mockOnRouteChange = (newRoute) => {
    route = newRoute;
}

describe('StartScreen.js', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<StartScreen onRouteChange={mockOnRouteChange} />)
  })

  it('expects to render StartScreen component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('expects route to change on click', () => {
    wrapper.find('[className="formButton"]').simulate('click')
    expect(route).toEqual('signin')
  })

}) 