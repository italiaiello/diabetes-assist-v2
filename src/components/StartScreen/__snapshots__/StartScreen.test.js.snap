// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartScreen.js expects to render StartScreen component 1`] = `ShallowWrapper {}`;
